src/components/LoginFormComponent.tsx

```diff
@@ -0,0 +1,92 @@
+import { TextField, Button } from "@material-ui/core";
+import { useCallback, useContext, useState } from "react";
+import styled from "styled-components";
+
+import { TranslationContext } from "../contexts/TranslationContext";
+import { loginAction } from "../reducers/authReducer";
+import { loginRequest, registerRequest } from "../services/authService";
+import { useAppDispatch } from "../store/hooks";
+
+const FormContainer = styled.form`
+  display: grid;
+  gap: 15px;
+  padding: 15px;
+  width: 50%;
+`;
+
+export default function LoginFormComponent() {
+  const dispatch = useAppDispatch();
```

Comment:
Please, use some form-related library,
Maintain the validation at least at some level (currently there's no way for user to even understand whether his credentials where accepted or not)

Please, use redux to maintain the login flow and state itself.